src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/FacebookPostAnalyzer.scala

```diff
@@ -1,26 +1,28 @@
 package com.microsoft.partnercatalyst.fortis.spark.analyzer
 
-import java.time.Instant.now
-import java.util.UUID.randomUUID
-
 import com.github.catalystcode.fortis.spark.streaming.facebook.dto.FacebookPost
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
 import com.microsoft.partnercatalyst.fortis.spark.transforms.image.ImageAnalyzer
 
 @SerialVersionUID(100L)
-class FacebookAnalyzer extends Analyzer[FacebookPost] with Serializable
+class FacebookPostAnalyzer extends Analyzer[FacebookPost] with Serializable with Loggable
   with AnalysisDefaults.EnableAll[FacebookPost] {
   override def toSchema(item: FacebookPost, locationFetcher: LocationFetcher, imageAnalyzer: ImageAnalyzer): ExtendedDetails[FacebookPost] = {
     ExtendedDetails(
-      id = randomUUID(),
-      createdAtEpoch = now.getEpochSecond,
+      eventid = item.post.getId,
+      eventtime = item.post.getUpdatedTime.getTime,
       body = item.post.getMessage,
-      title = "",
-      publisher = "Facebook",
-      sourceUrl = item.post.getPermalinkUrl.toString,
+      title = item.post.getCaption,
+      externalsourceid = item.post.getSource.toString,
+      pipelinekey = "Facebook",
       sharedLocations = Option(item.post.getPlace).map(_.getLocation) match {
         case Some(location) => locationFetcher(location.getLatitude, location.getLongitude).toList
-        case None => List()
+        case None =>
+          val errorMsg = "Empty PageIds argument for Facebook connector stream"
+          logFatalError(errorMsg)
+          throw new IllegalArgumentException(errorMsg)
```

Comment:
Will this get logged somehow (I suppose it'll print to stderr on the driver)? If not, might make sense to pass to the overload of `logFatalError` that takes `Throwable`.